core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -370,6 +383,20 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
 
       purged
     }
+
+    def forceComplete(): Int = {
```

Comment:
Looking further into it - i think this already works without having to do anything. Correct me if i'm wrong, but when a leadership change happens `ReplicaManager.makeFollowers` will get called. Which does:
```
partitionsToMakeFollower.foreach { partition =>
        val topicPartitionOperationKey = new TopicPartitionOperationKey(partition.topicPartition)
        tryCompleteDelayedProduce(topicPartitionOperationKey)
        tryCompleteDelayedFetch(topicPartitionOperationKey)
      }
```
At which point the `DelayedProduce` will error out with `Errors.NOT_LEADER_FOR_PARTITION` which will be translated to `Errors.NOT_COORDINATOR` in `TransactionStateManger.appendTransactionToLog`. So I think there is perhaps nothing to do